src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/playback/GerritMissedEventsPlaybackManager.java

```diff
@@ -554,4 +571,80 @@ public String getDisplayName() {
         return StringUtil.getDefaultDisplayNameForSpecificServer(this, getServerName());
     }
 
+    /**
+     * Responsible for persisting timestamps to xml.
+     * Time slices jump by 1000ms so the thread only checks every 1 second.
+     */
+    static class GerritMissedEventsPlaybackPersistRunnable implements Runnable {
+        private static String serverName;
+        private Thread worker;
+        private final AtomicBoolean running = new AtomicBoolean(false);
+        private static final long CHECK_INTERVAL = 1000;
+
+        /**
+         * Constructor.
+         * @param serverName name of server to persist timestamps for.
+         */
+        public GerritMissedEventsPlaybackPersistRunnable(String serverName) {
+            GerritMissedEventsPlaybackPersistRunnable.serverName = serverName;
+        }
+
+        /**
+         * Start the persistence thread loop.
+         */
+        public synchronized void start() {
+            if (!isRunning()) {
+                running.set(true);
+                worker = new Thread(this);
```

Comment:
There is `jenkins.util.Timer.get()` for these sorts of things, I'm debating with myself if that one can handle once a second... It probably would I think.

Reply:
I use jenkins scheduler to trigger the runnable every second.